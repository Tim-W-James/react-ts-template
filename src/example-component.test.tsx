// @vitest-environment happy-dom
import Link from "example-component";
import renderer from "react-test-renderer";

const toJson = (component: renderer.ReactTestRenderer) => {
  const result = component.toJSON();
  expect(result).toBeDefined();
  expect(result).not.toBeInstanceOf(Array);

  return result as renderer.ReactTestRendererJSON;
};

test("Link changes the class when hovered", () => {
  const component = renderer.create(
    <Link page="https://reactjs.org/">React</Link>
  );
  let tree = toJson(component);
  expect(tree).toMatchSnapshot();

  // Manually trigger the callback
  tree.props["onMouseEnter"]();

  // Re-rendering
  tree = toJson(component);
  expect(tree).toMatchSnapshot();

  // Manually trigger the callback
  tree.props["onMouseLeave"]();
  // Re-rendering
  tree = toJson(component);
  expect(tree).toMatchSnapshot();
});
