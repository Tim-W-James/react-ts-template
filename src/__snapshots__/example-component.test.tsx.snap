// Vitest Snapshot v1

exports[`Link changes the class when hovered 1`] = `
<a
  className="normal"
  href="https://reactjs.org/"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  React
</a>
`;

exports[`Link changes the class when hovered 2`] = `
<a
  className="hovered"
  href="https://reactjs.org/"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  React
</a>
`;

exports[`Link changes the class when hovered 3`] = `
<a
  className="normal"
  href="https://reactjs.org/"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  React
</a>
`;
